src/components/ContentNode/TabNavigator.vue

```diff
@@ -71,6 +71,22 @@ export default {
       currentTitle: this.titles[0],
     };
   },
+  watch: {
+    titles(newVal, oldVal) {
+      if (newVal.length < oldVal.length) {
+        // set to the first tab if selected tab was removed
+        const removedTab = oldVal.find((tab, i) => newVal[i] !== tab);
+        if (removedTab === this.currentTitle) {
+          // eslint-disable-next-line prefer-destructuring
+          this.currentTitle = newVal[0];
+        }
+      } else {
+        // set to newly added/changed tab
+        const newTab = newVal.find(tab => !oldVal.includes(tab));
+        this.currentTitle = newTab || this.currentTitle;
+      }
+    },
+  },
```

Comment:
Can this logic be simplified to something like:

1. If no `currentTitle` or the `currentTitle` is not part of the existing `titles`, just set the first one.
2. If `currentTitle` is in the `titles`, just do nothing.